src/api/schema/topology.rs

```diff
@@ -29,44 +32,232 @@ pub struct SourceData {
 #[derive(Clone)]
 pub struct Source(SourceData);
 
-lazy_static! {
-    static ref SOURCES: Arc<RwLock<Vec<Source>>> = Arc::new(RwLock::new(vec![]));
-}
-
 #[Object]
 impl Source {
     /// Source name
     async fn name(&self) -> String {
         self.0.name.clone()
     }
 
-    /// The output type given by the source
+    /// Source output type
     async fn output_type(&self) -> SourceOutputType {
         self.0.output_type.into()
     }
+
+    /// Transform outputs
+    async fn transforms(&self) -> Vec<Transform> {
+        filter_topology(|(_name, topology)| match topology {
+            Topology::Transform(t) if t.0.inputs.contains(&self.0.name) => Some(t.clone()),
+            _ => None,
+        })
+    }
+
+    //noinspection DuplicatedCode
+    /// Sink outputs
+    async fn sinks(&self) -> Vec<Sink> {
+        filter_topology(|(_name, topology)| match topology {
+            Topology::Sink(s) if s.0.inputs.contains(&self.0.name) => Some(s.clone()),
+            _ => None,
+        })
+    }
+}
+
+#[derive(Clone)]
+pub struct TransformData {
```

Comment:
These `SinkData` and `SourceData` types seem like they'd fit well as a `Data<T>` ?

Reply:
I kept them separate as it’s possible they’ll diverge as requirements evolve, but also maybe not. I’ll unify for now and split out later if needed 👍